src/ORM/DB.php

```diff
@@ -109,6 +126,124 @@ public static function get_conn($name = 'default')
         return null;
     }
 
+    /**
+     * Only use the primary database connection for the rest of the current request
+     *
+     * @internal
+     */
+    private static bool $mustUsePrimary = false;
+
+    /**
+     * Used by DB::withPrimary() to count the number of times it has been called
+     * Uses an int instead of a bool to allow for nested calls     *
+     *
+     * @internal
+     */
+    private static int $withPrimaryCount = 0;
+
+    /**
+     * The key of the previously choosen random replica config
+     *
+     * @internal
+     */
+    private static string $randomReplicaConfigKey = '';
+
+    /**
+     * Flag to determine if replicas can be used
```

Comment:
```suggestion
     * Whether replicas can be used
```
"flag" is a bit of a weird term here.

Reply:
Have removed functionality